App/SB/components/CommentBox/CommentBox.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { View, TextInput, Text, TouchableOpacity } from 'react-native'
+
+import Icons from '../../../Components/Icons'
+
+import styles from './statics/styles'
+
+type Props = {
+  keyboardHeight: number
+  value: string | undefined
+  onUpdate: (text: string) => void
+  onSubmit: () => void
+}
+
+const CommentBox = (props: Props) => {
+  const { keyboardHeight, value, onUpdate, onSubmit } = props
+
+  return (
+    <View style={[styles.commentFooter, { bottom: keyboardHeight }]}>
+      <TextInput
+        style={styles.textInput}
+        placeholder='Write a comment...'
+        placeholderTextColor='#4a4a4a'
+        onChangeText={onUpdate}
+        value={value}
```

Comment:
It's cool, somehow RN implements some components to be two-way-bound... `onChangeText` allows you to get updates, update your redux state or whatever, but `value` allows you to set the current text in the text input. We use it here to clear out the text input when our `authoringComment` state becomes `undefined`. Somehow they avoid a recursive death loop of events.